example/lib/board_editor_page.dart

```diff
@@ -0,0 +1,165 @@
+import 'package:board_example/board_theme.dart';
+import 'package:flutter/material.dart';
+import 'package:chessground/chessground.dart';
+import 'package:dartchess/dartchess.dart' as dc;
+import 'package:collection/collection.dart';
+
+class BoardEditorPage extends StatefulWidget {
+  const BoardEditorPage({super.key});
+
+  @override
+  State<StatefulWidget> createState() => _BoardEditorPageState();
+}
+
+class _BoardEditorPageState extends State<BoardEditorPage> {
+  Pieces pieces = readFen(dc.kInitialFEN);
+
+  Piece? pieceToAddOnTap = null;
+  bool deleteOnTap = false;
+
+  @override
+  Widget build(BuildContext context) {
+    final double screenWidth = MediaQuery.of(context).size.width;
+
+    final PieceSet pieceSet = PieceSet.merida;
+
+    final settings = BoardEditorSettings(
+      pieceAssets: pieceSet.assets,
+      colorScheme: BoardTheme.blue.colors,
+      enableCoordinates: true,
+    );
+    final boardEditor = BoardEditor(
+      size: screenWidth,
+      orientation: Side.white,
+      pieces: pieces,
+      settings: settings,
+      onTappedSquare: (squareId) => setState(() {
+        if (deleteOnTap) {
+          pieces.remove(squareId);
+        } else if (pieceToAddOnTap != null) {
+          pieces[squareId] = pieceToAddOnTap!;
+        }
+      }),
+      onDiscardedPiece: (squareId) => setState(() {
+        pieces.remove(squareId);
+      }),
+      onDroppedPiece: (origin, destination, piece) => setState(() {
+        pieces[destination] = piece;
+        if (origin != null) {
+          pieces.remove(origin);
+        }
+      }),
+    );
+
+    final makePieceMenu = (side) => PieceMenu(
+          side: side,
+          pieceSet: pieceSet,
+          squareSize: boardEditor.squareSize,
+          settings: settings,
+          selectedPiece: pieceToAddOnTap,
+          pieceTapped: (role) => setState(() {
+            pieceToAddOnTap = Piece(role: role, color: side);
+            deleteOnTap = false;
+          }),
+          deleteSelected: deleteOnTap,
+          deleteTapped: () => setState(() {
+            pieceToAddOnTap = null;
+            deleteOnTap = !deleteOnTap;
+          }),
+        );
+
+    return Scaffold(
+      appBar: AppBar(
+        title: const Text('Board Editor'),
+      ),
+      body: Center(
+        child: Column(
+          mainAxisAlignment: MainAxisAlignment.spaceEvenly,
+          children: [
+            makePieceMenu(Side.white),
+            boardEditor,
+            makePieceMenu(Side.black),
+            Text('FEN: ${writeFen(pieces)}'),
+          ],
+        ),
+      ),
+    );
+  }
+}
+
+class PieceMenu extends StatefulWidget {
```

Comment:
There is no associated state, so I think it can be converted to a `StatelessWidget`.